pkg/pillar/hypervisor/kvm.go

```diff
@@ -459,6 +454,39 @@ func (ctx kvmContext) Task(status *types.DomainStatus) types.Task {
 	}
 }
 
+// TBD: Have a better way to calculate this number.
+// For now it is based on some trial-and-error experiments.
+func minVMMOverhead(config types.DomainConfig) int64 {
+	return 600 << 20
+}
+
+func vmmOverhead(config types.DomainConfig,
+	globalConfig *types.ConfigItemValueMap) (int64, error) {
+	var overhead int64
+
+	// Global node setting has a higher priority
+	if globalConfig != nil {
+		VmmOverheadOverrideCfgItem, ok := globalConfig.GlobalSettings[types.VmmMemoryLimitInMiB]
+		if !ok {
+			return 0, logError("Missing key %s", string(types.VmmMemoryLimitInMiB))
+		}
+		if VmmOverheadOverrideCfgItem.IntValue > 0 {
+			overhead = int64(VmmOverheadOverrideCfgItem.IntValue) << 20
+		}
+	}
+	if overhead == 0 {
+		// XXX: This looks ugly and probably incorrect, please FIXME
```

Comment:
Why do you think this is ugly and "probably" incorrect? Do we really need to set an overhead when it's zero? Why not simply doesn't allow any overhead if it's set to 0?

Reply:
I agree and that's why I think we should not configure any minimum overhead. Do you have an idea how often this overhead is really used?

Reply:
We use it by default. We provide the guarantees to the customer, that when he deploys N VMs, those N VMs:
a) won't bloat over some threshold (that's why VMM maxmem, or "overhead", or "limit"), when it is bloated - OOM will do its job.
b) VM 'A' is guaranteed to have N amount of memory and VM 'B' can't affect the VM 'A' by over-consumption. 
c) and ideally we don't support over-commitment, so you can't start more VMs than you have RAM on your node (in reality we still can, but this can be fixed in nearest future, at least was discussed)

Reply:
Ok, understood. I still believe default value should be zero since it's configurable.... but I've got your point, you can let the "ugly" 2.5% for now...

Reply:
In the past we have based these heuristics on other systems which use qemu+kvm. Does the 2.5% come from some such other place?

Reply:
I'm not aware of such an estimation. Very suspicious to me :)